modules/tooling-support/src/main/java/org/mule/runtime/module/tooling/api/artifact/DeclarationSession.java

```diff
@@ -77,6 +77,14 @@
    */
   MetadataResult<ComponentMetadataTypesDescriptor> resolveComponentMetadata(ComponentElementDeclaration component);
 
+  /**
+   * Disposes the {@link org.mule.runtime.api.metadata.MetadataCache} associated to the component metadata resolution which is
+   * provided in order to store resources cross resolution by resolvers.
+   *
+   * @param component the component whose has a dynamic metadata types.
+   */
+  void disposeMetadataCache(ComponentElementDeclaration component);
```

Comment:
Shouldn't it be at the ToolingClient level? If the cache is shared between sessions

Reply:
Yes, it is shared between session but cannot be accessed without a Session :(.
Why? Because we need to get the key from component, either by its resolver or by its configuration/connection data.